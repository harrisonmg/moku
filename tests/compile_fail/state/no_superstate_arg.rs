use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    #[superstate()]
    impl State<BlinkyState> for Bottom {}
}

fn main() {}
