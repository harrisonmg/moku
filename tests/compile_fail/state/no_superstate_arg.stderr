error: the `moku::superstate` attribute requires a single State name as an argument, e.g. `#[superstate(Top)]`
  --> tests/compile_fail/state/no_superstate_arg.rs:13:5
   |
13 |     #[superstate()]
   |     ^
